gui/rendering/WireframeObject.cpp

```diff
@@ -0,0 +1,199 @@
+
+    m_indices.push_back(f.vertexIndices[1] - 1);
+    m_indices.push_back(f.vertexIndices[2] - 1);
+
+    m_indices.push_back(f.vertexIndices[2] - 1);
+    m_indices.push_back(f.vertexIndices[0] - 1);
+  }
+
+  std::vector<uint8_t> cArr = {(uint8_t)m_color.red(), (uint8_t)m_color.green(),
+                               (uint8_t)m_color.blue(),
+                               (uint8_t)m_color.alpha()};
+
+  for (auto v : m_vertices) {
+    auto vArr = v.ToArray();
+    m_verticesBuffer.insert(m_verticesBuffer.end(), vArr.begin(), vArr.end());
+    m_colorBuffer.insert(m_colorBuffer.end(), cArr.begin(), cArr.end());
+  }
+
+  CreateVBO();
+}
+
+rendering::WireframeObject::~WireframeObject(void) { DestroyVBO(); }
+
+void rendering::WireframeObject::SetColor(QColor color) {
+  m_color = color;
+  UpdateVBOColor();
+}
+
+rendering::SMesh rendering::WireframeObject::GetMesh() { return m_mesh; }
+
+void rendering::WireframeObject::UpdateVBOColor() {
+
+  m_openGLContext->makeCurrent(m_openGLContext->surface());
+
+  int size = m_colorBuffer.size() / 4;
+
+  std::vector<uint8_t> cArr = {(uint8_t)m_color.red(), (uint8_t)m_color.green(),
+                               (uint8_t)m_color.blue(),
+                               (uint8_t)m_color.alpha()};
+
+  m_colorBuffer.clear();
+
+  for (int iter = 0; iter < size; iter++)
+    m_colorBuffer.insert(m_colorBuffer.end(), cArr.begin(), cArr.end());
+
+  GLsizeiptr colorBufferSize = m_colorBuffer.size() * sizeof(uint8_t);
+
+  m_vao->bind();
+
+  glBindBuffer(GL_ARRAY_BUFFER, m_colorBufferId);
+  glBufferData(GL_ARRAY_BUFFER, colorBufferSize, m_colorBuffer.data(),
+               GL_DYNAMIC_DRAW);
+  glVertexAttribPointer(1, 4, GL_UNSIGNED_BYTE, GL_FALSE, 0, nullptr);
+  glEnableVertexAttribArray(1);
+}
+
+void rendering::WireframeObject::CreateVBO(void) {
+
+  m_openGLContext->makeCurrent(m_openGLContext->surface());
+
+  GLsizeiptr vboSize = m_verticesBuffer.size() * sizeof(GLfloat);
+  GLsizeiptr colorBufferSize = m_colorBuffer.size() * sizeof(uint8_t);
+  GLsizeiptr indexBufferSize = m_indices.size() * sizeof(GLuint);
+
+  m_vao = std::unique_ptr<QOpenGLVertexArrayObject>(
+      new QOpenGLVertexArrayObject(nullptr));
+  m_vao->create();
+  m_vao->bind();
+
+  glGenBuffers(1, &m_vbo);
+  CheckOpenGLError("glGenBuffers");
+  glBindBuffer(GL_ARRAY_BUFFER, m_vbo);
+  CheckOpenGLError("glBindBuffer");
+  glBufferData(GL_ARRAY_BUFFER, vboSize, m_verticesBuffer.data(),
+               GL_STATIC_DRAW);
+  CheckOpenGLError("glBufferData");
+  glVertexAttribPointer(0, 4, GL_FLOAT, GL_FALSE, 0, nullptr);
+  CheckOpenGLError("glVertexAttribPointer");
+  glEnableVertexAttribArray(0);
+  CheckOpenGLError("glEnableVertexAttribArray");
+
+  glGenBuffers(1, &m_colorBufferId);
+  CheckOpenGLError("glGenBuffers");
+  glBindBuffer(GL_ARRAY_BUFFER, m_colorBufferId);
+  CheckOpenGLError("glBindBuffer");
+  // glBufferData(GL_ARRAY_BUFFER, colorBufferSize, m_colorBuffer.data(),
+  // GL_STATIC_DRAW);
```

Comment:
to delete?

Reply:
I left it intentionally, for debugging, but it can be deleted.